app/services/data/get-all-claim-data.js

```diff
@@ -27,51 +27,39 @@ module.exports = function (reference, eligibilityId, claimId, status) {
-  return knex('ExtSchema.Eligibility')
-    .first()
-    .where({'Reference': reference, 'EligibilityId': eligibilityId, 'Status': status})
-    .then(function (eligibility) {
-      return eligibility
-    })
+function getEligilibility (schema, reference, eligibilityId) {
+  return knex(`${schema}.Eligibility`).first().where({'Reference': reference, 'EligibilityId': eligibilityId})
 }
 
-function getClaim (claimId, status) {
-  return knex('ExtSchema.Claim')
-    .first()
-    .where({'ClaimId': claimId, 'Status': status})
-    .then(function (claim) {
-      if (!claim) {
-        throw new Error(`Could not find valid completed Claim for claimId: ${claimId}`)
-      }
-      return claim
-    })
+function getClaim (schema, claimId) {
+  return knex(`${schema}.Claim`).first().where({'ClaimId': claimId})
 }
 
-function getPrisoner (reference, eligibilityId) {
-  return knex('ExtSchema.Prisoner').first().where({'Reference': reference, 'EligibilityId': eligibilityId})
+function getPrisoner (schema, reference, eligibilityId) {
+  return knex(`${schema}.Prisoner`).first().where({'Reference': reference, 'EligibilityId': eligibilityId})
 }
 
-function getVisitor (reference, eligibilityId) {
-  return knex('ExtSchema.Visitor').first().where({'Reference': reference, 'EligibilityId': eligibilityId})
+function getVisitor (schema, reference, eligibilityId) {
+  return knex(`${schema}.Visitor`).first().where({'Reference': reference, 'EligibilityId': eligibilityId})
 }
 
-function getClaimExpenses (claimId) {
-  return knex('ExtSchema.ClaimExpense').select().where({'ClaimId': claimId, 'IsEnabled': true})
+function getClaimExpenses (schema, claimId) {
+  return knex(`${schema}.ClaimExpense`).select().where({'ClaimId': claimId, 'IsEnabled': true})
 }
 
-function getClaimBankDetail (claimId) {
-  return knex('ExtSchema.ClaimBankDetail').first().where('ClaimId', claimId)
+function getClaimBankDetail (schema, claimId) {
+  return knex(`${schema}.ClaimBankDetail`).first().where('ClaimId', claimId)
 }
 
-function getClaimChildren (claimId) {
-  return knex('ExtSchema.ClaimChild').select().where({'ClaimId': claimId, 'IsEnabled': true})
+function getClaimChildren (schema, claimId) {
+  return knex(`${schema}.ClaimChild`).select().where({'ClaimId': claimId, 'IsEnabled': true})
 }
 
-function getClaimDocuments (claimId) {
-  return knex('ExtSchema.ClaimDocument').select().where({'ClaimId': claimId, 'IsEnabled': true})
+function getClaimDocuments (schema, claimId) {
+  return knex(`${schema}.ClaimDocument`).select().where({'ClaimId': claimId, 'IsEnabled': true})
 }
 
-function getEligibilityVisitorUpdateContactDetail (reference, eligibilityId) {
-  return knex('ExtSchema.EligibilityVisitorUpdateContactDetail').first().where({'Reference': reference, 'EligibilityId': eligibilityId}).orderBy('DateSubmitted', 'desc')
+function getEligibilityVisitorUpdateContactDetail (schema, reference, eligibilityId) {
+  if (schema === 'IntSchema') return Promise.resolve(null)
```

Comment:
Could perhaps invert this check ie. if (schema === 'ExtSchema') { //do stuff } else { return Promise.resolve(null) }

I think it's slightly more readable, and we're explicitly checking for the external schema before doing anything, instead of doing nothing only when IntSchema is passed in